src/tracers/blob_saver_tracer.rs

```diff
@@ -0,0 +1,112 @@
+use super::tracer::Tracer;
+use crate::{
+    eravm_error::{EraVmError, HeapError},
+    state::VMState,
+    value::FatPointer,
+    Opcode,
+};
+use std::collections::HashMap;
+use u256::{H160, H256, U256};
+use zkevm_opcode_defs::ethereum_types::Address;
+use zkevm_opcode_defs::sha2::{Digest, Sha256};
+
+#[derive(Default, Debug)]
+pub struct BlobSaverTracer {
+    pub blobs: HashMap<U256, Vec<U256>>,
+}
+
+impl BlobSaverTracer {
+    pub fn new() -> Self {
+        Self {
+            blobs: Default::default(),
+        }
+    }
+}
+
+const CONTRACT_DEPLOYER_ADDRESS: Address = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x80, 0x06,
+]);
+
+const KNOWN_CODES_STORAGE_ADDRESS: Address = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x80, 0x04,
+]);
+
+// Hardcoded signature of `publishEVMBytecode` function.
+// In hex is 0x964eb607
+const PUBLISH_BYTECODE_SIGNATURE: [u8; 4] = [0x96, 0x4e, 0xb6, 0x7];
+
+pub(crate) fn hash_evm_bytecode(bytecode: &[u8]) -> H256 {
```

Comment:
is there any reason we need `pub(crate)` here?

Reply:
Done